manifests/daemon.pp

```diff
@@ -45,6 +45,8 @@
 #  Service startup scripts style (e.g. rc, upstart or systemd).
 #  Can also be set to `none` when you don't want the class to create a startup script/unit_file for you.
 #  Typically this can be used when a package is already providing the file.
+# @param unit_after
+#  List of units that should be started prior to this daemon. Only supported for the systemd init_style.
```

Comment:
every exporter class inherits from prometheus::daemon. You need to add this parameter to those classes as well (including documentation)

Reply:
I really only use the define for creating my own exporters, so I had intended on letting it ride as an under-the-radar addition that the individual exporters can implement as needed (it may not even make sense for all of them). I do see the desire to expose the feature up higher though, so I'll try to get that change added here in the next little while.